crates/oxc_isolated_declarations/src/scope.rs

```diff
@@ -51,48 +53,36 @@ impl<'a> ScopeTree<'a> {
     }
 
     pub fn has_reference(&self, name: &str) -> bool {
-        let Some(scope) = self.levels.last() else { unreachable!() };
-        scope.value_references.contains(name) || scope.type_references.contains(name)
-    }
-
-    fn add_value_binding(&mut self, ident: Atom<'a>) {
-        let scope = self.levels.last_mut().unwrap();
-        scope.value_bindings.insert(ident);
-    }
-
-    fn add_type_binding(&mut self, ident: Atom<'a>) {
-        let scope = self.levels.last_mut().unwrap();
-        scope.type_bindings.insert(ident);
+        let scope = self.levels.last().unwrap();
```

Comment:
What happens when two symbols with the same name are declared and merged?

```ts
interface Foo {}
namespace Foo {}
```

Reply:
In declarations, they don't merge, all of it will print. See https://www.typescriptlang.org/play/?target=99#code/JYOwLgpgTgZghgYwgAgGIHt3IN4F8BQIcAthAM4AOiKGWe++EAHhelGDmpsrkA